src/main/java/org/ladocuploader/app/preparers/HouseholdDetailsPreparer.java

```diff
@@ -45,6 +50,23 @@ public Map<String, SubmissionField> prepareSubmissionFields(Submission submissio
 
         var uuid = householdMember.get("uuid");
         results.put("householdUSCitizen" + i, new SingleField("householdUSCitizenDerived", nonCitizens != null && nonCitizens.contains(uuid) ? "No" : "Yes", i + 1));
+
+        List<String> raceInput = (List)submission.getInputData().get(RACE_PREFIX + uuid + "[]");
+        String ethnicityInput = (String) submission.getInputData().get(ETHNICITY_PREFIX + uuid);
+        String raceEthnicCode = "";
+
+        if (raceInput != null && !raceInput.isEmpty()) {
+          raceEthnicCode = raceInput.stream()
+              .map(t -> RaceType.getAbbreviationFromValue(t))
+              // 'AN' maps to both American Indian and Alaskan Native, if both are chosen, only keep one 'AN'
+              .distinct()
+              .collect(Collectors.joining(","));
+        }
+
+        if (ethnicityInput != null && !ethnicityInput.isEmpty()) {
+          raceEthnicCode += "/" + EthnicityType.getAbbreviationFromValue(ethnicityInput);
+        }
+        results.put("householdRaceEthnicCode" + i, new SingleField("householdRaceEthnicCode", raceEthnicCode, i+1));
       }
     }
     return results;
```

Comment:
This is more for me: curious about whether it makes sense to make each of these inout checks into a method separate from the main method. Just noticing that it's getting pretty big. I could also see a good reason to split it out into methods for testing purposes.

Reply:
I did refactor the code a little here.